java-source/src/main/java/com/example/flow/ExampleFlow.java

```diff
@@ -66,57 +66,57 @@ public Initiator(IOUState iou, Party otherParty) {
+            // Obtain a reference to the notary we want to use.
+            final Party notary = single(getServiceHub().getNetworkMapCache().getNotaryNodes()).getNotaryIdentity();
+
+            // Stage 1.
+            progressTracker.setCurrentStep(GENERATING_TRANSACTION);
+            // Generate an unsigned transaction.
+            final Command txCommand = new Command(new IOUContract.Commands.Create(), iou.getParticipants());
+            final TransactionBuilder unsignedTx = new TransactionType.General.Builder(notary).withItems(iou, txCommand);
+
+            // Stage 2.
+            progressTracker.setCurrentStep(VERIFYING_TRANSACTION);
+            // Verify that the transaction is valid.
             try {
-                // Prep.
-                // Obtain a reference to our key pair. Currently, the only key pair used is the one which is registered with
-                // the NetWorkMapService. In a future milestone release we'll implement HD key generation such that new keys
-                // can be generated for each transaction.
-                final KeyPair keyPair = getServiceHub().getLegalIdentityKey();
-                // Obtain a reference to the notary we want to use.
-                final Party notary = single(getServiceHub().getNetworkMapCache().getNotaryNodes()).getNotaryIdentity();
-
-                // Stage 1.
-                progressTracker.setCurrentStep(GENERATING_TRANSACTION);
-                // Generate an unsigned transaction.
-                final Command txCommand = new Command(new IOUContract.Commands.Create(), iou.getParticipants());
-                final TransactionBuilder unsignedTx = new TransactionType.General.Builder(notary).withItems(iou, txCommand);
-
-                // Stage 2.
-                progressTracker.setCurrentStep(VERIFYING_TRANSACTION);
-                // Verify that the transaction is valid.
                 unsignedTx.toWireTransaction().toLedgerTransaction(getServiceHub()).verify();
-
-                // Stage 3.
-                progressTracker.setCurrentStep(SIGNING_TRANSACTION);
-                final SignedTransaction partSignedTx = unsignedTx.signWith(keyPair).toSignedTransaction(false);
-
-                // Stage 4.
-                progressTracker.setCurrentStep(SENDING_TRANSACTION);
-                // Send the state across the wire to the designated counterparty.
-                // -----------------------
-                // Flow jumps to Acceptor.
-                // -----------------------
-                this.send(otherParty, partSignedTx);
-
-                return new ExampleFlowResult.Success(String.format("Transaction id %s committed to ledger.", partSignedTx.getId()));
-
-            } catch(Exception ex) {
-                // Just catch all exception types.
-                return new ExampleFlowResult.Failure(ex.getMessage());
+            } catch (AttachmentResolutionException|TransactionVerificationException|TransactionResolutionException ex) {
```

Comment:
All three exceptions are now FlowExceptions so you don't need to do this try-catch-rethrow.